### PR TITLE
initial 5.0 commit

### PR DESCRIPTION
**Pull request by deepchoudhery**
- Version.props changes
- efcore upgrades

### CODE REVIEW

File: src/VS.Web.CG.EFCore/ModelMetadata.cs [hunk 1]
```diff
@@ -133,7 +133,7 @@ public INavigationMetadata[] Navigations
                 if (_navigations == null)
                 {
                     _navigations = EntityType.GetNavigations()
-                        .Where(n => n.IsDependentToPrincipal() == true && n.ForeignKey.Properties.All(p => !p.IsShadowProperty()))
+                        .Where(n => n.IsOnDependent == true && n.ForeignKey.Properties.All(p => !p.IsShadowProperty()))
```
Comment (vijayrkn) [inline src/VS.Web.CG.EFCore/ModelMetadata.cs]: @bricelam - Is this the right replacement function?

File: test/VS.Web.CG.MSBuild.Test/ProjectReferenceInformationProviderTests.cs [hunk 2]
```diff
@@ -30,7 +30,7 @@ public class ProjectReferenceInformationTests
   <ItemGroup>
     <ProjectReference Include=""..\Library2\Library2.csproj"" />
   </ItemGroup>
-  <ItemGroup Condition=""'$(TargetFramework)' == 'net461' "">
+  <ItemGroup Condition=""'$(TargetFramework)' == 'netcoreapp5.0' "">
```
Comment (vijayrkn) [inline test/VS.Web.CG.MSBuild.Test/ProjectReferenceInformationProviderTests.cs]: why not net5.0?

File: test/TestApps/ModelTypesLocatorTestWebApp/ModelTypesLocatorTestWebApp.csproj [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <Project Sdk="Microsoft.NET.Sdk.Web">
   <PropertyGroup>
-    <TargetFrameworks>netcoreapp3.0</TargetFrameworks>
+    <TargetFrameworks>netcoreapp5.0</TargetFrameworks>
```
Comment (vijayrkn) [inline test/TestApps/ModelTypesLocatorTestWebApp/ModelTypesLocatorTestWebApp.csproj]: cant you use net5.0 in all cases?

File: test/Shared/MSBuildProjectStrings.cs [hunk 4]
```diff
@@ -661,7 +660,7 @@ protected override void OnModelCreating(ModelBuilder builder)
     </When>
     <Otherwise>
       <PropertyGroup >
-        <EvaluateProjectInfoForCodeGenerationAssemblyPath>$(MSBuildThisFileDirectory)..\toolAssets\net461\Microsoft.VisualStudio.Web.CodeGeneration.Msbuild.dll</EvaluateProjectInfoForCodeGenerationAssemblyPath>
+        <EvaluateProjectInfoForCodeGenerationAssemblyPath>$(MSBuildThisFileDirectory)..\toolAssets\netcoreapp5.0\Microsoft.VisualStudio.Web.CodeGeneration.Msbuild.dll</EvaluateProjectInfoForCodeGenerationAssemblyPath>
```
Comment (vijayrkn) [inline test/Shared/MSBuildProjectStrings.cs]: same here. Why not net5.0?

File: test/Shared/MSBuildProjectStrings.cs [hunk 5]
```diff
@@ -950,7 +949,7 @@ public class Blog
     </When>
     <Otherwise>
       <PropertyGroup >
-        <EvaluateProjectInfoForCodeGenerationAssemblyPath>$(MSBuildThisFileDirectory)\toolAssets\net461\Microsoft.VisualStudio.Web.CodeGeneration.Msbuild.dll</EvaluateProjectInfoForCodeGenerationAssemblyPath>
+        <EvaluateProjectInfoForCodeGenerationAssemblyPath>$(MSBuildThisFileDirectory)\toolAssets\netcoreapp5.0\Microsoft.VisualStudio.Web.CodeGeneration.Msbuild.dll</EvaluateProjectInfoForCodeGenerationAssemblyPath>
```
Comment (vijayrkn) [inline test/Shared/MSBuildProjectStrings.cs]: here too.

File: test/TestApps/PreviousPackage/PreviousPackage.csproj [hunk 6]
```diff
@@ -1,8 +1,9 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk.Web">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFramework>netcoreapp5.0</TargetFramework>
```
Comment (vijayrkn) [inline test/TestApps/PreviousPackage/PreviousPackage.csproj]: here too.

File: test/TestApps/StablePackage/StablePackage.csproj [hunk 7]
```diff
@@ -1,8 +1,9 @@
 <Project Sdk="Microsoft.NET.Sdk.Web">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.0</TargetFramework>
+    <TargetFramework>netcoreapp5.0</TargetFramework>
```
Comment (vijayrkn) [inline test/TestApps/StablePackage/StablePackage.csproj]: +1

File: test/VS.Web.CG.MSBuild.Test/ProjectReferenceInformationProviderTests.cs [hunk 8]
```diff
@@ -19,7 +19,7 @@ public class ProjectReferenceInformationTests
     <RootNamespace>Microsoft.Library</RootNamespace>
     <ProjectName>Library1</ProjectName>
     <OutputType>Library</OutputType>
-    <TargetFrameworks>netstandard2.0;net461</TargetFrameworks>
+    <TargetFramework>net5.0</TargetFramework>
```
Comment (vijayrkn) [inline test/VS.Web.CG.MSBuild.Test/ProjectReferenceInformationProviderTests.cs]: +1

File: src/VS.Web.CG.EFCore/ModelMetadata.cs (hunk 1 above)
Comment (deepchoudhery) [reply src/VS.Web.CG.EFCore/ModelMetadata.cs]: it is as per https://github.com/dotnet/efcore/blob/master/src/EFCore/Extensions/NavigationExtensions.cs

**Pull request merged**